nuget/helpers/lib/NuGetUpdater/NuGetUpdater.Core/Analyze/NuGetContext.cs

```diff
@@ -142,11 +142,22 @@ public static string[] GetPackageSourceUrls(string currentDirectory)
             }
 
             var metadataResource = await sourceRepository.GetResourceAsync<PackageMetadataResource>(cancellationToken);
-            var metadata = await metadataResource.GetMetadataAsync(packageIdentity, SourceCacheContext, Logger, cancellationToken);
-            var url = metadata.ProjectUrl ?? metadata.LicenseUrl;
-            if (url is not null)
+            if (metadataResource is not null)
             {
-                return url.ToString();
+                try
+                {
+                    var metadata = await metadataResource.GetMetadataAsync(packageIdentity, SourceCacheContext, Logger, cancellationToken);
+                    var url = metadata.ProjectUrl ?? metadata.LicenseUrl;
+                    if (url is not null)
+                    {
+                        return url.ToString();
+                    }
+                }
+                catch (ArgumentException)
+                {
+                    // there was an issue deserializing the package metadata; this doesn't necessarily mean the package
+                    // is unavailable, just that we can't return a URL
```

Comment:
Should this be logged?

Reply:
I considered it, but the case where this happened was data coming directly from api.nuget.org (specifically [this](https://api.nuget.org/v3/registration5-gz-semver2/castle.core/index.json) line 37) so if they're not enforcing a valid data shape then we probably don't care either.